app/lib/frontend/model_properties.dart

```diff
@@ -20,7 +29,7 @@ class Pubspec {
         _json = jsonMap;
 
   factory Pubspec.fromYaml(String yamlString) =>
-      new Pubspec.fromJson(loadYaml(yamlString) as Map<String, dynamic>);
+      new Pubspec.fromJson(_loadYaml(yamlString));
```

Comment:
.fromYaml accepts a string while .fromJson accepts a map? That's a bit inconsistent. I prefer to have .fromJson do JSON decoding and having a .fromMap that constructs the object from a decoded object.

Reply:
Yeah, that is unfortunate, but all of the JSON serialization code is using `factory ClassName.fromJson(Map<String, dynamic)` as their deserialization constructor, and I'm not sure if the [code generator](https://pub.dartlang.org/packages/json_serializable) would work with a different signature. (Also they use `Map toJson()` instead of `toMap()` and `String toJson()` :( )

Reply:
Although, `fromJson(Map)` may be acceptable if its pair is `parseJson(String)`...